src/components/EditableField/styles/EditableField.Input.css.ts

```diff
@@ -34,51 +37,95 @@ export const ComponentUI = styled('div')`
   z-index: 1;
   border-bottom: 1px solid ${COLOURS.invisible};
 
-  .is-disabled &:hover {
+  .${STATES_CLASSNAMES.isLarge} & {
+    height: ${field.height.large};
+  }
+
+  .${STATES_CLASSNAMES.isDisabled} &:hover {
     cursor: initial;
     border-bottom: 1px solid ${COLOURS.invisible};
   }
 
-  .is-active & {
+  .${STATES_CLASSNAMES.isActive} & {
     pointer-events: none;
     z-index: 2;
     border-bottom-color: ${COLOURS.invisible} !important;
   }
+
+  &.${STATES_CLASSNAMES.isInline} {
+    position: relative;
+  }
 `
 
 export const InputWrapperUI = styled('div')`
   position: relative;
-  height: ${CONTENT_HEIGHT - 2}px;
+  height: ${input.height.medium};
   width: 100%;
 
-  .has-options & {
+  .${STATES_CLASSNAMES.isLarge} & {
+    height: ${field.height.large};
+  }
+
+  .${STATES_CLASSNAMES.hasOptions} & {
     width: calc(100% - 70px);
   }
 
-  .is-empty & {
+  .${STATES_CLASSNAMES.isEmpty} & {
     width: 100%;
   }
+
+  &::before {
+    opacity: 0;
+    will-change: transform, font-size;
+    transition-property: transform, font-size;
+    transition-timing-function: linear;
+    transition-duration: 0.2s;
+    pointer-events: none;
+  }
+
+  .${STATES_CLASSNAMES.hasActiveFields} & {
+    &::before {
+      content: ${({ placeholder }) => `"${placeholder}"`};
```

Comment:
I really like that solution to pass the custom placeholder. Another good example of how great styled component could be!